### PR TITLE
WIP: Schema mod

### PR DESCRIPTION
**Pull request by CJ-Wright**
This may be a rather rough PR.

This is my attempt to merge some of the existing DBs/work done by the Billinge group with the existing Regolith DBs. I've tried to keep the commits separate so we can cherry pick as needed. I hope there will be some discussion about where everything should go (into the main schema or into the user defined schemas from the RC) and the proper data models.

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Should the proposals and grants collections be merged?

**Comment by CJ-Wright** [comment]
Attn: @sbillinge

**Comment by scopatz** [comment]
This has conflicts in the tests now

File: regolith/candpbuilder.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    return start < time.time() < end
+
+
+def is_pending(sd, sm, sy):
+    s = '{}/{}/{}'.format(sd, sm, sy)
+    start = time.mktime(datetime.datetime.strptime(s, "%d/%m/%Y").timetuple())
+    return time.time() < start
+
+
+class CPBuilder(BuilderBase):
```
Comment (scopatz) [inline regolith/candpbuilder.py]: This should probably be a subclass of the latex builder once it is merged.

File: regolith/commands.py [hunk 2]
```diff
@@ -136,3 +136,13 @@ def yaml_to_json(rc):
...
+
+
+def validate(rc):
+    """Validate the combined database against the schemas"""
+    from regolith.schemas import validate
+    print('=' * 10 + '\nVALIDATING\n')
+    for name, collection in rc.client.chained_db.items():
+        for doc_id, doc in collection.items():
+            validate(name, doc)
+    print('\nNO ERORS IN DBS\n' + '=' * 15)
```
Comment (scopatz) [inline regolith/commands.py]: Maybe less screaming, please.  Non-blocking.

File: regolith/migration.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+"""Maps between old keys and new"""
+
+m = {'proposals': {'day': 'due_day',
```
Comment (scopatz) [inline regolith/migration.py]: This is not a good variable name.

**Pull request closed without merging**